### PR TITLE
8292717: Clean up checking of testing requirements in configure

### PR DESCRIPTION
**Pull request by GoeLin**
I backport this because then 8314495 can be backported clean. Also brings 11 closer to 17.

configure.ac::
I had to resolve two chunks. In one, I needed to omit the coding for JMH which is not
supported by 11.

jdk-options.m4, toolchanin.m4
Deleted function has different body, deleted by hand.

lib-tests.m4: Copyright.

The original change moves and renames three
functions to lib-tests.m4. It does not change
the implementation of these.
So I modified the added code in lib-tests.m4
to have the unchanged function bodies that
were deleted in the other files.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] [JDK-8292717](https://bugs.openjdk.org/browse/JDK-8292717) needs maintainer approval
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8292717](https://bugs.openjdk.org/browse/JDK-8292717): Clean up checking of testing requirements in configure (**Bug** - P3 - Approved)

### Reviewers
 * [Aleksey Shipilev](https://openjdk.org/census#shade) (@shipilev - **Reviewer**) ⚠️ Review applies to [2b85a35e](https://git.openjdk.org/jdk11u-dev/pull/2600/files/2b85a35ed11d798ca4edde4247c68ba0f622bb3c)
 * [Martin Doerr](https://openjdk.org/census#mdoerr) (@TheRealMDoerr - **Reviewer**) ⚠️ Review applies to [2b85a35e](https://git.openjdk.org/jdk11u-dev/pull/2600/files/2b85a35ed11d798ca4edde4247c68ba0f622bb3c)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk11u-dev.git pull/2600/head:pull/2600` \
`$ git checkout pull/2600`

Update a local copy of the PR: \
`$ git checkout pull/2600` \
`$ git pull https://git.openjdk.org/jdk11u-dev.git pull/2600/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 2600`

View PR using the GUI difftool: \
`$ git pr show -t 2600`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk11u-dev/pull/2600.diff">https://git.openjdk.org/jdk11u-dev/pull/2600.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk11u-dev/pull/2600#issuecomment-1999212196)

### CODE REVIEW

**Comment by GoeLin** [comment]
Hi @shipilev, 

I figured it works smoothest if I backport in the order they arrived in 17. 
So I handled this one now. You might want to have a look, should be familiar :)

Best, Goetz.

**Review by shipilev — APPROVED** [review]
This looks fine, thanks. It is generally hard to check that we have not introduced bugs while doing these move, but as long as they are mostly mechanical, I think we are safer.

I don't see the `JTREG_MINIMUM_VERSION=6.1` hunk in the original commit, but I guess we need to move it from `toolchain.m4` for 11u-dev.

File: make/autoconf/configure.ac [hunk 1]
```diff
@@ -233,7 +232,7 @@ HOTSPOT_SETUP_JVM_FEATURES
 ###############################################################################
 
 JDKOPT_DETECT_INTREE_EC
-JDKOPT_ENABLE_DISABLE_FAILURE_HANDLER
+LIB_TESTS_ENABLE_DISABLE_FAILURE_HANDLER
```
Comment (TheRealMDoerr) [inline make/autoconf/configure.ac:235]: Original change inserts an empty line here.

File: make/autoconf/lib-tests.m4 [hunk 2]
```diff
@@ -23,6 +23,13 @@
 # questions.
 #
 
+################################################################################
+# Setup libraries and functionalities needed to test the JDK.
+################################################################################
+
+# Minimum supported version
+JTREG_MINIMUM_VERSION=6.1
```
Comment (TheRealMDoerr) [inline make/autoconf/lib-tests.m4:31]: Moving the stuff here from the deleted file is probably ok.

**Review by TheRealMDoerr — APPROVED** [review]
I think it's good. Feel free to add the empty line.

**Comment by GoeLin** [comment]
GHA failure: StringRepeat.java: Could not reserve enough space for 2097152KB object heap

**Pull request closed without merging**